src/shogun/machine/IterativeLinearMachine.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVELINEARMACHINE_H__
+#define _ITERATIVELINEARMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/lib/SGVector.h>
+
+
+namespace shogun
+{
+  class CFeatures;
+  class CLabels;
+
+  class CIterativeLinearMachine : public CLinearMachine
+   {
+      public:
+        
+      /* default constructor */  
+       CIterativeLinearMachine();
+       virtual ~CIterativeLinearMachine();
+       
+       /** Train machine
+        * @params f features
+        * @params l labels 
+        * @return whether training was successful
+        */
+       virtual bool train(CDotFeatures* f, CLabels* l);
```

Comment:
We could enforce constness here with this new design, right? This can be done for all the method below (`continue_train`, `iteration`). @karlnapf

Reply:
Let's follow shogun's design for now which is to take features here

Reply:
(and change this later)